proposals/lip-0049.md

```diff
@@ -636,15 +644,17 @@ def verify(ccu: Transaction, ccm: CCM) -> None:
 ##### Execution
 
 ```python
-def execute(ccu: Transaction, ccm: CCM) -> None:
+def execute(ccu: CCU, ccm: CCM) -> None:
+    ccmParams = decode(sidechainTerminatedCCMParamsSchema, ccm.params)
+
     if terminatedStateAccount(ccm.chainID) exists:
         if terminatedStateAccount(ccm.chainID).initialized == True:
             return
-        terminatedStateAccount(trs.params.chainID).stateRoot = ccm.params.stateRoot
-        terminatedStateAccount(trs.params.chainID).mainchainStateRoot = EMPTY_HASH
-        terminatedStateAccount(trs.params.chainID).initialized = True
+        terminatedStateAccount(ccu.params.chainID).stateRoot = ccmParams.stateRoot
+        terminatedStateAccount(ccu.params.chainID).mainchainStateRoot = EMPTY_HASH
+        terminatedStateAccount(ccu.params.chainID).initialized = True
```

Comment:
Please check the lines above as there was a bug before (`trs` was not defined).

Reply:
`ccmParams.chainID` in lines 650, 651 and 653-655

Reply:
Now I saw that it was already changed. All good

Reply:
I assume this can now be resolved then.

Reply:
Just to be sure: will someone from SDK team double check that the implementation is indeed using the correct ID?

Reply:
As discussed, @ricott1 will still check this point.